crates/forge/tests/cli/test_cmd/fuzz.rs

```diff
@@ -0,0 +1,735 @@
+    }
+}
+   "#,
+    );
+    // Test should pass when replay failure with changed function signature.
+    cmd.forge_fuse().args(["test"]).assert_success().stdout_eq(str![[r#"
+[COMPILING_FILES] with [SOLC_VERSION]
+[SOLC_VERSION] [ELAPSED]
+Compiler run successful!
+
+Ran 1 test for test/Counter.t.sol:CounterTest
+[PASS] testFuzz_SetNumber(uint8) (runs: 256, [AVG_GAS])
+Suite result: ok. 1 passed; 0 failed; 0 skipped; [ELAPSED]
+
+Ran 1 test suite [ELAPSED]: 1 tests passed, 0 failed, 0 skipped (1 total tests)
+
+"#]]);
+
+    // Change test back to the original one that produced the counterexample.
+    prj.add_test(
+        "Counter.t.sol",
+        r#"
+import {Test} from "forge-std/Test.sol";
+
+contract CounterTest is Test {
+    function testFuzz_SetNumber(uint256 x) public pure {
+        require(x > 200);
+    }
+}
+   "#,
+    );
+    // Test should fail with replayed counterexample 2 (0 runs).
+    cmd.forge_fuse().args(["test"]).assert_failure().stdout_eq(str![[r#"
+...
+Failing tests:
+Encountered 1 failing test in test/Counter.t.sol:CounterTest
+[FAIL: EvmError: Revert; counterexample: calldata=0x5c7f60d70000000000000000000000000000000000000000000000000000000000000002 args=[2]] testFuzz_SetNumber(uint256) (runs: 0, [AVG_GAS])
+...
+
+"#]]);
+});
+
+forgetest_init!(fuzz_basic, |prj, cmd| {
+    prj.wipe_contracts();
+    prj.add_test(
+        "Fuzz.t.sol",
+        r#"
+import "forge-std/Test.sol";
+
+contract FuzzTest is Test {
+    constructor() {
+        emit log("constructor");
+    }
+
+    function setUp() public {
+        emit log("setUp");
+    }
+
+    function testShouldFailFuzz(uint8 x) public {
+        emit log("testFailFuzz");
+        require(x > 128, "should revert");
+    }
+
+    function testSuccessfulFuzz(uint128 a, uint128 b) public {
+        emit log("testSuccessfulFuzz");
+        assertEq(uint256(a) + uint256(b), uint256(a) + uint256(b));
+    }
+
+    function testToStringFuzz(bytes32 data) public {
+        vm.toString(data);
+    }
+}
+   "#,
+    );
+
+    cmd.args(["test"]).assert_failure().stdout_eq(str![[r#"
+...
+Ran 3 tests for test/Fuzz.t.sol:FuzzTest
+[FAIL: should revert; counterexample: calldata=[..] args=[..]] testShouldFailFuzz(uint8) (runs: [..], [AVG_GAS])
+[PASS] testSuccessfulFuzz(uint128,uint128) (runs: 256, [AVG_GAS])
+[PASS] testToStringFuzz(bytes32) (runs: 256, [AVG_GAS])
+Suite result: FAILED. 2 passed; 1 failed; 0 skipped; [ELAPSED]
+
+Ran 1 test suite [ELAPSED]: 2 tests passed, 1 failed, 0 skipped (3 total tests)
+
+Failing tests:
+Encountered 1 failing test in test/Fuzz.t.sol:FuzzTest
+[FAIL: should revert; counterexample: calldata=[..] args=[..]] testShouldFailFuzz(uint8) (runs: [..], [AVG_GAS])
+
+Encountered a total of 1 failing tests, 2 tests succeeded
+
+Tip: Run `forge test --rerun` to retry only the 1 failed test
+
+"#]]);
+});
+
+// Test that showcases PUSH collection on normal fuzzing.
+// Ignored until we collect them in a smarter way.
+forgetest_init!(
+    #[ignore]
```

Comment:
same as for https://github.com/foundry-rs/foundry/pull/12049#discussion_r2438755939 with AST seeded fuzz dictionary we will collect values of `0x4446` and `0x5556` and use to fuzz selectors.

For the
```
        counter += numToIncrement;
        counterX2 += numToIncrement * 2;
```
case wonder if folding could help https://github.com/foundry-rs/foundry/issues/12044 cc @0xrusowsky